### DISCUSSION TITLE
Filtering products by custom model field via swell-js

### DISCUSSION THREAD

**Discussion by logeshswell** [swellstores/community#9] [Help] [answered]
How to implement a usecase to fetch a list of food dishes available in a particular restaurant, where there can be 'm' restaurants each having 'n' food dishes?

**Comment by logeshswell** [comment, marked as answer]
1. Create a custom content model _Restaurant_ with at least 2 fields `slug` ("dominos") and `name` ("Domino's Pizza") along with auto-generated fields `id`, `date_created` and `date_updated`.
  * Model data could be accessible from the Swell frontend APIs (or swell-js) if it is created with the `content` namespace.
2. In _Product_ model containing the actual information of food dishes, create a custom lookup field (`restaurant_id`) that is mapped to the _restaurant_ model.
  * `restaurant_id` refers to the restaurant to which the food dish belongs.
3. You can use the following method to fetch the list of records:
```ts
import swell from 'swell-js'
swell.init('store_id', 'public_key')

queryParams = {
  restaurant_id: 'dominos_id_value',
  expand: 'restaurant'
}
products = await swell.products.list(queryParams)
```

**Comment by logeshswell** [comment]
### Can I use slug value ("dominos") instead of auto-generated id for querying products?

Yes, you can handle it in 2 ways.
1. You can either add a custom query for `restaurant_id` field.
2. Make `slug` field as a secondary field in _Restaurant_ model. [**Recommended**]

Please follow the below instructions for the recommended approach:

1. Navigate to _Restaurant_ Model using this link: https://store_id.swell.store/admin/settings/models/com.content_restaurant
2. Click **Actions** > **View in console**
3. Change it to **PUT** request type and add the below dict in **Body**.
```json
{
  "secondary_field": "slug"
}
```
5. Reload the swell dashboard and navigate to _Product_ model using this link: https://store_id.swell.store/admin/settings/models/com.products
6. Choose **Edit** button of `restaurant_id` field and click **Advanced options** toggle from the popup.
7. Choose `Secondary (slug)` from **Lookup to field** dropdown and click **Save** button.

Now, you can use the slug value instead of auto-generated id to query the products of your choice:
```ts
queryParams = {
  restaurant_id: 'dominos', // slug value
  expand: 'restaurant'
}
products = await swell.products.list(queryParams)
```

**Reply by dirums** [reply]
I've followed the steps but cannot query the products by slug. Please help.
